src/pages/case-studies.js

```diff
@@ -1,38 +1,261 @@
+      name: 'IT_Service_Modernization',
+      isChecked: false,
+    },
+    {
+      id: 3,
+      value: 'Product & Design',
+      name: 'Product_Design',
+      isChecked: false,
+    },
+    {
+      id: 4,
+      value: 'Security & Compliance',
+      name: 'Security_Compliance',
+      isChecked: false,
+    },
+    { id: 5, value: 'Data Services', name: 'Data_Services', isChecked: false },
+    {
+      id: 6,
+      value: 'Workforce Development',
+      name: 'Workforce_Development',
+      isChecked: false,
+    },
+  ];
+
+  const [filterState, setFilterState] = useState([]);
+  let checked;
+  // Handle onChange from Filters
+  const callback = (e) => {
+    checked = e.target.checked;
+    console.log(checked);
+    if (e.target.name !== 'undefined') {
+
+      if (!(filterState.includes(e.target.name))) {
+        // If the filter has not been selected, add it to the array of selected items.
+        setFilterState([...filterState, e.target.name]);
+        checked = false;
+      } else {
+        // If the filter has already been selected, remove it from the array
+        filterState.splice(filterState.indexOf(e.target.name), 1)
+        setFilterState([...filterState]);
+        checked = true;
+      }
+    }
+  };
+
+  // Re-renders when filter is clicked and state is updated because useState is async.
+  useEffect(() => { }, [filterState.length]);
+
+
+  let cases;
+  const caseStudies = data.allStrapiCaseStudy.nodes;
+
+  if (caseStudies) {
+    // If no filters are selected show all case studies
+    if (!filterState.length) {
+      cases = caseStudies;
+    } else {
+      cases = caseStudies.filter((caseStudy) => {
+        let category = caseStudy.Service_Category[0].Category;
+        // For each case study, loop through the selected filters and only 
+        // return results that match the selected categories.
+        for (let filter of filterState) {
+          if (category === filter) {
+            return true
+          }
+        }
+      });
+    }
+  }
+
   return (
     <RedLayout>
-        <Hero
-            title='Work that makes a difference'
-            description='Our work impacts the daily lives of millions of people. See how we’ve helped agencies build resilient services at scale.'
-        />
-        <Helmet>
-            <title data-react-helmet="true">Case Study Landing page</title>
-        </Helmet>
-      <pre>{JSON.stringify(data, null, 4)}</pre>
+      <Helmet>
+        <title data-react-helmet='true'>Case Studies Landing page</title>
+      </Helmet>
+      <Hero
+        title='Work that makes a difference'
+        description='Our work impacts the daily lives of millions of people. See how we’ve helped agencies build resilient services at scale.'
+      />
+      <section className='section--case-studies--filter'>
+        <h2 className='body'>Filter by service category</h2>
+        <div className={`checkboxes ${filterState}`}>
+          {categories.map((item, index) => (
```

Comment:
When mapping through an array, you don't need to pass the index to get each item in the array. `categories[index].id` can be replaced by `item.id`.

Also, when mapping through an array/object, react needs a `key` attribute. More info here: https://reactjs.org/docs/lists-and-keys.html